stubs/tree-sitter/tree_sitter/binding.pyi

```diff
@@ -0,0 +1,95 @@
+from _typeshed import Incomplete
+from typing import ClassVar
+from typing_extensions import final
+
+from tree_sitter import Language
+
+@final
+class Node:
+    @property
+    def start_byte(self) -> int: ...
+    @property
+    def start_point(self) -> tuple[int, int]: ...
+    @property
+    def end_byte(self) -> int: ...
+    @property
+    def end_point(self) -> tuple[int, int]: ...
+    @property
+    def has_changes(self) -> bool: ...
+    @property
+    def has_error(self) -> bool: ...
+    @property
+    def is_missing(self) -> bool: ...
+    @property
+    def is_named(self) -> bool: ...
+    @property
+    def child_count(self) -> int: ...
+    @property
+    def named_child_count(self) -> bool: ...
+    @property
+    def children(self) -> list[Node]: ...
+    @property
+    def next_named_sibling(self) -> Node | None: ...
+    @property
+    def next_sibling(self) -> Node | None: ...
+    @property
+    def parent(self) -> Node | None: ...
+    @property
+    def prev_named_sibling(self) -> Node | None: ...
+    @property
+    def prev_sibling(self) -> Node | None: ...
+    @property
+    def text(self) -> bytes: ...
+    @property
+    def type(self) -> str: ...
+    __hash__: ClassVar[None]  # type: ignore[assignment]
+    def child_by_field_id(self, __id: int) -> Node | None: ...
+    def child_by_field_name(self, __name: str) -> Node | None: ...
+    def sexp(self) -> str: ...
+    def walk(self) -> TreeCursor: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    # There are __ge__, __gt__, __le__, __lt__ methods but they seem to always return False
+    #
+    #    >>> n
+    #    <Node kind=call, start_point=(0, 0), end_point=(0, 14)>
+    #    >>> n >= "", n <= "", n >= 0, n <= 0, n >= (0,0), n <= (0,0)
+    #    (False, False, False, False, False, False)
+
+@final
+class Parser:
+    # At runtime, Parser(1, 2, 3) ignores the arguments, but that's most likely buggy code
+    def __init__(self) -> None: ...
+    def parse(self, source: bytes, old_tree: Tree | None = ..., keep_text: bool = ...) -> Tree: ...
+    def set_language(self, __language: Language) -> None: ...
+
+@final
+class Query:
+    def captures(self, *args, **kwargs): ...
+    def matches(self, *args, **kwargs): ...
+
+@final
+class Tree:
+    @property
+    def root_node(self) -> Node: ...
+    @property
+    def text(self) -> bytes: ...
```

Comment:
This will be pretty annoying to check in practice, because it is a union return type that is usually just `bytes`. Maybe:

```suggestion
    def text(self) -> bytes | Any: ...  # technically ReadableBuffer | Any
```